glean_parser/go_server.py

```diff
@@ -120,29 +127,31 @@ def output_go(
                         + " metric type."
                     )
                     continue
+
                 for ping in metric.send_in_pings:
-                    if ping != "events":
-                        (
-                            print(
-                                "❌ Non-events ping reference found."
-                                + PING_METRIC_ERROR_MSG
-                                + f"Ignoring the {ping} ping type."
-                            )
-                        )
-                        continue
+                    if metric.type == "event" and metric not in event_metrics:
+                        event_metrics.append(metric)
+
                     metrics_by_type = ping_to_metrics[ping]
                     metrics_list = metrics_by_type.setdefault(metric.type, [])
                     metrics_list.append(metric)
 
-    if "event" not in ping_to_metrics["events"]:
-        print("❌ No event metrics found...at least one event metric is required")
+    PING_METRIC_ERROR_MSG = (
+        " Server-side environment is simplified and this"
+        + " parser doesn't generate individual metric files. Make sure to pass all"
+        + " your ping and metric definitions in a single invocation of the parser."
+    )
+    if not ping_to_metrics:
+        print("❌ No pings with metrics found." + PING_METRIC_ERROR_MSG)
         return
 
     extension = ".go"
     filepath = output_dir / ("server_events" + extension)
     with filepath.open("w", encoding="utf-8") as fd:
         fd.write(
             template.render(
-                parser_version=__version__, events_ping=ping_to_metrics["events"]
+                parser_version=__version__,
```

Comment:
I noticed that the `server_go*.go` files in tests/data don't have current version, maybe __version__ isn't being set? (They also don't have the leading `v`, so that might be incorrect)

Reply:
That's the pattern in the test data output so it's not hardcoded; generating those files to compare to it has something like `glean_parser v15.0.2.dev7+gd4b1b802.d20241016` which is manually replaced with `{current_version}` to create these files.

The tests swap that string with whatever the current version is
```
    glean_version = f"glean_parser v{glean_parser.__version__}"
    # use replace instead of format since Go uses { }
    compare = compare_raw.replace("{current_version}", glean_version)
```

Reply:
Looking more closely, this originally came from the ruby implementation that I based the go one off of; but none of the other ones have these tests to validate the produced code against a "golden file".. we could consider removing this kind of test, since there are tests which generate the code & perform calls to verify the log output produced.

https://github.com/mozilla/glean_parser/blob/main/tests/test_ruby_server.py#L79-L99

Reply:
IIRC we didn't have integration tests that called the generated code and validated schema back when the Ruby test was added. I think it would be fine to get rid of `_compare` tests but it's fine to keep them too. I was planning to refactor and standardize server parsers at some point so will probably address that then.